db/sql/table_compiled_template.sql

```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS compiled_template;
+CREATE TABLE compiled_template
+(
+    template_id     CHAR(36)      NOT NULL   COMMENT 'FOREIGN KEY',
+    body            MEDIUMTEXT    NOT NULL   COMMENT 'the template compiled with helpers and partials',
+    created_at      DATETIME      NOT NULL   DEFAULT NOW() COMMENT 'UTC datetime',
+
+    UNIQUE KEY (template_id),
```

Comment:
this should be PK

Reply:
according to @ccopto this is the way to go ^^

Reply:
@ccopto no pk?

Reply:
Throwing me under the bus eh @danmcadams :trollface:

Long Answer: it is possible to have a PK and a FK in the same column? yes, does it bring any benefit? no (actually if the data is joined regularly it will behave a little slower than having altogether in just one table). The reasoning for having a FK is so we can allow a dependency on another table's column, strictly speaking, the FK column on table 2 is just a reference to table 1 column and not the owner of the record (this is just a design argument).

Short answer: it doesn't matter, FK+Unique is a better design with little benefit.

Reply:
:busstop: